### PR TITLE
Trm 21659/back button

### PR DESCRIPTION
**Pull request by xwatkins**
Use url to manage state of:
- query string
- facets
- sort column
- sort direction

### CODE REVIEW

File: src/results/ResultsContainer.tsx [hunk 1]
```diff
@@ -118,25 +138,113 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+    } = this.getURLParams(queryParamFromUrl);
+
+    const index = selectedFacets.findIndex(
+      selectedFacet =>
+        selectedFacet.name === facetName && selectedFacet.value === facetValue
+    );
+
+    this.setURLParams(
+      query,
+      [...selectedFacets.slice(0, index), ...selectedFacets.slice(index + 1)],
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: To be more clear about what is this going on, this line also can be written as:

     // Before calling this.setURLParamas
     const updatedSelectedFacets = selectedFacets.splice(index, 1);
     // Within this.setURLParams
     updatedSelectedFacets,

File: src/results/ResultsContainer.tsx [hunk 2]
```diff
@@ -118,25 +138,113 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+      sortColumn,
+      sortDirection
+    );
+  };
+
+  updateColumnSort = (column: SortableColumn) => {
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing return type.

File: src/results/utils/utils.ts [hunk 3]
```diff
@@ -17,19 +17,15 @@ const getAPIQueryUrl = (
   queryString: string,
   columns: string[],
   selectedFacets: SelectedFacet[],
-  sortBy: SortableColumns | undefined = undefined,
-  sortDirectionKey: keyof SortDirectionsType | undefined = SortDirections.ascend
-    .app as keyof SortDirectionsType
+  sortColumn: SortableColumn | undefined = undefined,
+  sortDirection: SortDirection | undefined = SortDirection.ascend
```
Comment (its-mahdi) [inline src/results/utils/utils.ts]: Missing comma in the end of the line.

File: src/results/utils/utils.ts [hunk 4]
```diff
@@ -17,19 +17,15 @@ const getAPIQueryUrl = (
...
   return getQueryUrl(
     `${queryString}${facetsQueryString}`,
     columns,
-    undefined,
-    sortBy,
-    sortBy && sortBy in SortableColumns
-      ? SortDirections[sortDirectionKey].api
-      : undefined
+    sortColumn,
+    getApiSortDirection(SortDirection[sortDirection])
```
Comment (its-mahdi) [inline src/results/utils/utils.ts]: Missing comma in the end of the line.

File: src/utils/apiUrls.ts [hunk 5]
```diff
@@ -51,16 +55,12 @@ export const getSuggesterUrl = (url: string, value: string) =>
 export const getQueryUrl = (
   encodedQueryString: string,
   columns: string[],
-  cursor?: string,
-  sortBy?: SortableColumns | undefined,
-  sortDirection?: string | undefined
+  sortBy?: SortableColumn | undefined,
+  sortDirection?: SortDirectionApi | undefined
```
Comment (its-mahdi) [inline src/utils/apiUrls.ts]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 6]
```diff
@@ -50,56 +41,50 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
   constructor(props: ResultsProps) {
     super(props);
     this.state = { selectedRows: {} };
-    this.handleRowSelect = this.handleRowSelect.bind(this);
   }
 
   componentDidMount() {
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing return type.

File: src/results/ResultsContainer.tsx [hunk 7]
```diff
@@ -50,56 +41,50 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+    const {
+      query,
+      selectedFacets,
+      sortColumn,
+      sortDirection,
+    } = this.getURLParams(queryParamFromUrl);
     dispatchClearResults();
     dispatchFetchBatchOfResultsIfNeeded(
-      getAPIQueryUrl(queryString, columns, selectedFacets, column, direction)
+      getAPIQueryUrl(query, columns, selectedFacets, sortColumn, sortDirection)
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 8]
```diff
@@ -50,56 +41,50 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
     );
   }
 
   componentDidUpdate(prevProps: ResultsProps) {
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing return type.

File: src/results/ResultsContainer.tsx [hunk 9]
```diff
@@ -50,56 +41,50 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+      } = this.getURLParams(queryParamFromUrl);
       dispatchClearResults();
       dispatchFetchBatchOfResultsIfNeeded(
-        getAPIQueryUrl(queryString, columns, selectedFacets, column, direction)
+        getAPIQueryUrl(
+          query,
+          columns,
+          selectedFacets,
+          sortColumn,
+          sortDirection
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 10]
```diff
@@ -109,7 +94,42 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+      sortColumn: sortColumn && SortableColumn[sortColumn],
+      sortDirection: sortDirection && SortDirection[sortDirection],
+    };
+  };
+
+  setURLParams = (
+    query: string,
+    selectedFacets: SelectedFacet[],
+    sortColumn: SortableColumn,
+    sortDirection: SortDirection
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 11]
```diff
@@ -118,25 +138,113 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+      sortDirection,
+    } = this.getURLParams(queryParamFromUrl);
+
+    const facet: SelectedFacet = { name: facetName, value: facetValue };
+
+    this.setURLParams(
+      query,
+      [...selectedFacets.slice(0, selectedFacets.length), facet],
+      sortColumn,
+      sortDirection
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 12]
```diff
@@ -118,25 +138,113 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+    const index = selectedFacets.findIndex(
+      selectedFacet =>
+        selectedFacet.name === facetName && selectedFacet.value === facetValue
+    );
+
+    this.setURLParams(
+      query,
+      [...selectedFacets.slice(0, index), ...selectedFacets.slice(index + 1)],
+      sortColumn,
+      sortDirection
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 13]
```diff
@@ -118,25 +138,113 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
-      dispatchUpdateColumnSort,
       dispatchFetchBatchOfResultsIfNeeded,
       namespace,
-      sort,
       nextUrl,
       totalNumberResults,
     } = this.props;
     const { selectedRows } = this.state;
+    const { selectedFacets, sortColumn, sortDirection } = this.getURLParams(
+      queryParamFromUrl
```
Comment (its-mahdi) [inline src/results/ResultsContainer.tsx]: Missing comma in the end of the line.

File: src/results/ResultsContainer.tsx [hunk 14]
```diff
@@ -109,7 +94,42 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
     dispatchReset();
   }
 
-  handleRowSelect(rowId: string) {
+  getURLParams = (url: string) => {
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/results/ResultsContainer.tsx]: This seems to be inferred by TS?

File: src/results/ResultsContainer.tsx [hunk 15]
```diff
@@ -109,7 +94,42 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+    const urlParams = queryStringModule.parse(url);
+    const { query, facets, sort, dir } = urlParams;
+
+    let selectedFacets;
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/results/ResultsContainer.tsx]: Dine

File: src/results/ResultsContainer.tsx [hunk 16]
```diff
@@ -109,7 +94,42 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
...
+  setURLParams = (
+    query: string,
+    selectedFacets: SelectedFacet[],
+    sortColumn: SortableColumn,
+    sortDirection: SortDirection
+  ) => {
+    const { history } = this.props;
+    history.push({
+      pathname: '/uniprotkb',
+      search: `query=${query}${this.facetsAsString(selectedFacets)}${
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/results/ResultsContainer.tsx]: That's nice

**Pull request merged**